ddtrace/tracer.py

```diff
@@ -667,13 +663,8 @@ def set_service_info(self, *args, **kwargs):
         """Set the information about the given service."""
         return
 
-    def wrap(
-        self,
-        name=None,  # type: Optional[str]
-        service=None,  # type: Optional[str]
-        resource=None,  # type: Optional[str]
-        span_type=None,  # type: Optional[str]
-    ):
+    def wrap(self, name=None, service=None, resource=None, span_type=None):
+        # type: (Optional[str], Optional[str], Optional[str], Optional[str]) -> Callable
```

Comment:
The more detailed we can get with `Callable` the better.

e.g. 
```suggestion
        # type: (Optional[str], Optional[str], Optional[str], Optional[str]) -> Callable[[Callable[[...], typing.Any], Callable[[...], typing.Any]
```

(I think that is right, since we return the `wrap_decorator` which takes a function with any arguments and any return type, and returns a function which takes any arguments with any return type.... ?)